widget/embedded/src/hooks/useReferrer.ts

```diff
@@ -0,0 +1,42 @@
+import type { LoadingStatus } from '../store/meta';
+
+import { useWallets } from '@rango-dev/wallets-react';
+import { WalletTypes } from '@rango-dev/wallets-shared';
+import { useEffect, useRef } from 'react';
+
+type UseReferrerParams = {
+  referrer: string | null;
+  loadingStatus: LoadingStatus;
+};
+
+export function useReferrer(params: UseReferrerParams): void {
```

Comment:
You can make this hook more general : 
```tstype forceAutoConnectParams = {
  isActiveTab: boolean;
  loadingStatus: MetaState['loadingStatus'];
  walletType: WalletTypes
};
function useforceAutoConnect(params: ForceAutoConnectParams): void {
  const { isActiveTab, loadingStatus, walletType} = params;

  const { connect, state } = useWallets();
  const initiated = useRef(false);

  const walletState = state(walletType);

  useEffect(() => {
    const shouldTryConnect =
      isActiveTab &&
      loadingStatus === 'success' &&
      walletState &&
      walletState.installed &&
      !walletState.connecting &&
      !walletState.connected;

    if (shouldTryConnect && !initiated.current) {
      initiated.current = true;
      connect(walletType)
        .then()
        .catch((error: any) => {
          console.error(error);
        });
    }
  }, [walletState, isActiveTab, loadingStatus]);
}

useForceAutoConnect(WalletTypes.SAFE)

useForceAutoConnect(reffererWalletType)
```

Reply:
It seems that isActiveTab isn't implemented in the widget